frontend/src/pages/CourseSelector/CourseDescription/CourseDescription.jsx

```diff
@@ -85,12 +85,14 @@ const CourseDescription = () => {
       }
     };
 
+    const fetchCourseData = async () => {
+      await Promise.all([getCourse(), getPathToCoursesById(id), getCourseCapacityById(id)]);
+      setPageLoaded(true);
```

Comment:
ah so that was the bug - i made a hotfix but couldnt find the root cause. Nice job spotting it!